### PR TITLE
Check quota before transfer ownership

### LINKED ISSUE #15664: Transfer ownership should check quota
When issuing a transfer ownership from userA to userB. The command should check if the quota of userB is sufficient.

If not the command should not even be started and a message should be shown.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #15664

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: apps/files/lib/Command/TransferOwnership.php [hunk 1]
```diff
@@ -174,6 +178,15 @@ private function walkFiles(View $view, $path, \Closure $callBack) {
 	 */
 	protected function analyse(OutputInterface $output) {
 		$view = new View();
+
+		$output->writeln('Validating quota');
+		$size = $view->getFileInfo($this->sourcePath, false)->getSize(false);
+		$freeSpace = $view->free_space($this->destinationUser . '/files/');
+		if ($size > $freeSpace) {
```
Comment (juliushaertl) [inline apps/files/lib/Command/TransferOwnership.php]: Maybe also check for negative values e.g. when transferring files to a user that has not logged in yet the size is not computed and therefore the free space check fails.

**Review by juliusknorr — APPROVED** [review]
Besides a possible enhancement, tested and works

File: apps/files/lib/Command/TransferOwnership.php (hunk 1 above)
Comment (rullzer) [reply apps/files/lib/Command/TransferOwnership.php]: Next time;)

**Pull request merged**